src/ConsoleArgumentBag.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console;
+
+final class ConsoleArgumentBag
+{
+    /** @var ConsoleInputArgument[] */
+    protected array $arguments = [];
+
+    /** @var string[] */
+    protected array $path = [];
+
+    /**
+     * @param array<string|int, mixed> $arguments
+     */
+    public function __construct(array $arguments)
+    {
+        $this->path = array_filter([
+            $arguments[0] ?? null,
+            $arguments[1] ?? null,
+        ]);
+
+        unset($arguments[0], $arguments[1]);
+
+        foreach (array_values($arguments) as $position => $argument) {
+            if (str_starts_with($argument, '--')) {
+                [$key, $value] = $this->parseNamedArgument($argument);
+
+                $this->add(
+                    new ConsoleInputArgument(
+                        name: $key,
+                        value: $value,
+                        position: $position,
+                    )
+                );
+
+                continue;
```

Comment:
Code style preference here, but if two blocks are of "equal value", I prefer to use `if/else` instead of early returns or continues.

Not it looks like the part within this `if` block is an edge case that's being handled first, while that's definitely not so. 

Another remark is that I think it might be better to move the construction logic to the `ConsoleInputArgument` itself. Something like this:

```php
$this->add(ConsoleInputArgument::fromRaw($argument));
```

`fromRaw` is a sucky name though, but I can't think of any better alternatives atm

Reply:
I renamed that to fromString lemme know if that's fine